### DISCUSSION TITLE
Updating manually Entities

### DISCUSSION THREAD

**Discussion by radaniba** [marklogic-community/envision#249] [Q&A]
Hi there,

When we change Entities attributes in DHCC, are they temporarily stored somewhere before the deployement step ?

I added some relationship between entities that made two of my entities non editable from the UI for some reason (cannot unfold the menu) and was wondering if I can fix this by manually editing the JSON document created from DHCC ?

Rad

**Comment by wpaven** [comment]
Which version are you using?
Entities are written to the /entities directory in the data hub project folder.
Dragging a node in Connect will force the write. 

Hope this helps,
Pete

**Reply by radaniba** [reply]
Hi @wpaven thank you for getting back to me

I am using 5.4.1

I am not able to do what you suggested and looking at the log files I found this 

```
[http-nio-9003-exec-6] ERROR o.s.b.a.w.s.e.ErrorMvcAutoConfiguration$StaticView - Cannot render error page for request [/websocket/995/3uc5llla/htmlfile] and exception [Broken pipe] as the response has already been committed. As a result, the response may have the wrong status code.
```

and higher the error message is saying 

```
20:53:06.758 [http-nio-9003-exec-8] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is com.marklogic.client.FailedRequestException: Local message: failed to POST at /envision/model/toDatahub.sjs: Internal Server Error. Server Message: JS-JAVASCRIPT: (function (exports, module, require) { // cache buster: 6f519706-9537-43f1-b3eb-bffb693232aa -- Error running JavaScript request: RangeError: Maximum call stack size exceeded . See the MarkLogic server error log for further detail.] with root cause
com.marklogic.client.FailedRequestException: Local message: failed to POST at /envision/model/toDatahub.sjs: Internal Server Error. Server Message: JS-JAVASCRIPT: (function (exports, module, require) { // cache buster: 6f519706-9537-43f1-b3eb-bffb693232aa -- Error running JavaScript request: RangeError: Maximum call stack size exceeded . See the MarkLogic server error log for further detail.
```
When I login to the Quickstart and check the entities I can see them, but cannot edit them neither (the change not persistent) it is like they are locked after editing them in DHCC and making the deployment

**Reply by radaniba** [reply]
Problem solved, I had to open a new model, then load the model again and the errors are now gone, cache issues maybe ?

**Reply by wpaven** [reply]
Possibly cache.  But you can't use 5.4.x with QuickStart. (data hub 5.2.x)  You need to use CE 5.4.x with Data Hub Central 5.4.x

The entity model stays the same, but steps/flows changes in 5.4. So that can bork things if you're trying to use quickstart. 

Also, CE 5.4.1 has some bugs that have been fixed in 5.4.2. So I'd suggest using the latest. 
Hope this helps,
Pete

**Reply by radaniba** [reply]
Thanks @wpaven 

Yes I am using CE 5.4.1 with Quickstart (DH 5.4.1) so these are not compatible you say?

I will schedule for an upgrade to 5.4.2 thanks for the suggestion

Rad

**Reply by wpaven** [reply]
Yes, Hub Central updates the formats of how flows/steps are configured.  QuickStep works with the 5.2.x and earlier format.   Quickstart will be going away in 5.5.

But CE works with the new Hub Central format.  So use 5.4.x CE with Hub Central 5.4.x.

If you want to use QuickStart, then you'd use CE 2.0.5.  But you'll want to upgrade to hub central format because as the formats continue to evolve new capabilities will require the newer formats.

Hope this helps,
Pete
